pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,217 @@
+package ftp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"gofr.dev/pkg/gofr/datasource"
+
+	"github.com/jlaffaye/ftp"
+)
+
+// ftpFileSystem represents a file system interface over FTP.
+type ftpFileSystem struct {
+	file   *ftpFile          // Pointer to a single file object (not used in the provided methods)
+	conn   ServerConn        // FTP server connection
+	config *Config           // FTP configuration
+	logger datasource.Logger // Logger interface for logging
+}
+
+// Config represents the FTP configuration.
+type Config struct {
+	Host      string // FTP server hostname
+	User      string // FTP username
+	Password  string // FTP password
+	Port      string // FTP port
+	RemoteDir string // Remote directory path
+}
+
+// New initializes a new instance of ftpFileSystem with provided configuration.
+func New(config *Config) datasource.FileSystem {
+	return &ftpFileSystem{config: config}
+}
+
+// UseLogger sets the logger interface for the FTP file system.
+func (f *ftpFileSystem) UseLogger(logger interface{}) {
+	f.logger = logger.(datasource.Logger)
+}
+
+// UseMetrics sets the metrics for the ftpFileSystem client which asserts the Metrics interface.
+// Currently not implemented.
+func (*ftpFileSystem) UseMetrics(metrics interface{}) {
+
+}
+
+// Connect establishes a connection to the FTP server and logs in.
+func (f *ftpFileSystem) Connect() {
+	// Construct FTP server address
+	ftpServer := fmt.Sprintf("%v:%v", f.config.Host, f.config.Port)
+
+	// Connect to FTP server
+	const dialTimeoutSeconds = 5
+
+	conn, err := ftp.Dial(ftpServer, ftp.DialWithTimeout(dialTimeoutSeconds*time.Second))
```

Comment:
`dialTimeoutSeconds` should be specified as `const dialTimeout = 5 * time.Second` instead of scaling it up separately.